hypothesis-python/src/hypothesis/core.py

```diff
@@ -1154,6 +1164,12 @@ def run_engine(self):
                 err.__cause__ = err.__context__ = e
                 errors_to_report.append((fragments, err))
             except BaseException as e:
+                try:
+                    notes = info._expected_exception.__notes__
+                except AttributeError:
+                    pass
+                else:
+                    e.__notes__ = notes
```

Comment:
I think it's probably better to leave this out.  While seeing the notes might be helpful, I'm concerned that moving the notes from the old exception might be confusing (especially if the new exception already has notes!).

Did you have a particular case in mind where it would be useful?

Reply:
Happy to update this another way, but with a removal of this and no other change, the "positive" tests (where the note should be available) don't pass. It seems to be because the code path where the note's currently added is not walked by the reproduction (which directly calls `execute_one` rather than doing so through `_execute_once_for_engine`). Is this acceptable? Am I missing something that should be evident?

Reply:
That suggests to me that we might be adding the note in the wrong place... but I'm not sure exactly what the solution should be.  I'll take a look soon.

Reply:
I had this sense but wasn't sure how you'd prefer to alter the attachment strategy/placement, sorry about that. Looks good from your recent commit!